File: src/main/java/com/csci4050/bookstore/model/Address.java
package com.csci4050.bookstore.model;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

@Entity
@Table(name = "address")
public class Address {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  @Column(name = "id")
  private Integer id;

  @ManyToOne(optional = false, fetch = FetchType.LAZY)
  @JoinColumn(name = "user_id", referencedColumnName = "id")
  private User user;

  @Column(name = "name")
  private String name;

  @Column(name = "address_1")
  private String address1;

  @Column(name = "address_2")
  private String address2;

  @Column(name = "city")
  private String city;

  @Column(name = "state")
  private String state;

  @Column(name = "zip")
  private Integer zip;

  @Column(name = "country")
  private String country;

  @Column(name = "phone")
  private String phoneNumber;

  public Integer getId() {
    return this.id;
  }

  public void setUser(User user) {
    this.user = user;
  }

  public String getName() {
    return this.name;
  }

  public void setName(String name) {
    this.name = name;
  }

  public String getAddress1() {
    return this.address1;
  }

  public void setAddress1(String address1) {
    this.address1 = address1;
  }

  public String getAddress2() {
    return this.address2;
  }

  public void setAddress2(String address2) {
    this.address2 = address2;
  }

  public String getCity() {
    return this.city;
  }

  public void setCity(String city) {
    this.city = city;
  }

  public String getState() {
    return this.state;
  }

  public void setState(String state) {
    this.state = state;
  }

  public Integer getZip() {
    return this.zip;
  }

  public void setZip(Integer zip) {
    this.zip = zip;
  }

  public String getCountry() {
    return this.country;
  }

  public void setCountry(String country) {
    this.country = country;
  }

  public String getPhoneNumber() {
    return this.phoneNumber;
  }

  public void setPhoneNumber(String phoneNumber) {
    this.phoneNumber = phoneNumber;
  }
}

File: frontend/src/components/Address.js
import React from 'react';

import { Typography } from 'antd';

const { Text } = Typography;

function Address(props) {
  const {
    name,
    address1,
    address2,
    city,
    state,
    zip,
    country,
    phoneNumber,
  } = props;

  return (
    <div>
      <Text>
        <b>{name}</b>
      </Text>
      <br />
      <Text>{address1}</Text>
      <br />
      {address2 && [<Text key='a2'>{address2}</Text>, <br key='br1' />]}
      <Text>
        {city}, {state} {zip}
      </Text>
      <br />
      <Text>{country}</Text>
      <br />
      {phoneNumber && [<Text key='ph'>Phone number: {phoneNumber}</Text>, <br key='br2' />]}
    </div>
  );
}

export default Address;
